users/views.py

```diff
@@ -116,8 +118,29 @@ def get_context_data(self, **kwargs):
 
 class TunnistamoOidcAuthorizeView(AuthorizeView):
     def get(self, request, *args, **kwargs):
+        scope = request.GET.get('scope')
+        if scope:
+            query_params = request.GET.copy()
+            query_params['scope'] = self._add_api_scopes(scope)
+            request.GET = query_params
+
         language = request.GET.get('lang')
         if language and language in (l[0] for l in settings.LANGUAGES):
             with translation.override(language):
                 return super().get(request, *args, **kwargs)
+
         return super().get(request, *args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        scope = request.POST.get('scope')
+        if scope:
+            query_params = request.POST.copy()
+            query_params['scope'] = self._add_api_scopes(scope)
+            request.POST = query_params
```

Comment:
Could be DRYer, e.g.
```python
class TunnistamoOidcAuthorizeView(AuthorizeView): 
    def get(...):
        request.GET = _extend_scope_in_query_params(request.GET)
        ...

    def post(...):
        request.POST = _extend_scope_in_query_params(request.POST)
        ...


def _extend_scope_in_query_params(query_params):
    scope = query_params.get('scope')
    if scope:
        query_params = query_params.copy()
        query_params['scope'] = _add_api_scopes(scope)
    return query_params
```

Note: assumed _add_api_scopes to be moved outside of the class too. (I think free methods usually make more sense than staticmethods)